source/declarations.tex

```diff
@@ -2935,8 +2935,12 @@
   using C::x;
   int x(double);
 };
-int f(D* d) {
-  return d->x();    // ambiguous: \tcode{B::x} or \tcode{C::x}
+void f(D* d) {
+   d->x();       // ambiguous: \tcode{x}'s direct parent \tcode{A} is an ambiguous base of the naming class \tcode{D}
```

Comment:
agreed - that parent sinifies the container of the declaration - so in
struct A { void f(); }; - regardless of how member lookup finds 'f' in
whichever hierarchy - the direct parent would be 'A'.  Should I use direct
containing class?

Faisal Vali


On Wed, Jun 8, 2016 at 5:33 PM, Thomas Köppe <notifications@github.com>
wrote:

> In source/declarations.tex
> <https://github.com/cplusplus/draft/pull/726#discussion_r66352769>:
>
> > @@ -2935,8 +2935,12 @@
> >    using C::x;
> >    int x(double);
> >  };
> > -int f(D* d) {
> > -  return d->x();    // ambiguous: \tcode{B::x} or \tcode{C::x}
> > +void f(D* d) {
> > +   d->x();       // ambiguous: \tcode{x}'s direct parent \tcode{A} is an ambiguous base of the naming class \tcode{D}
>
> We never use the word "parent" in the context of classes. (We only use it
> in the context of threads and directories.) "Parent-child" is a *terrible*
> metaphor for class inheritance. A better metaphor is "base-derived".
>
> —
> You are receiving this because you were mentioned.
> Reply to this email directly, view it on GitHub
> <https://github.com/cplusplus/draft/pull/726/files/b79aef1dd96dd35e71094a5453967c5b74cf8646#r66352769>,
> or mute the thread
> <https://github.com/notifications/unsubscribe/AAxzPhiSgZHCtSD45VUKIW6EdEtj21rWks5qJ0M2gaJpZM4If6p6>
> .
>